tools/azure-pipelines/jobs-template.yml

```diff
@@ -0,0 +1,129 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+parameters:
+  test_pool_definition: # where is compiliation and unit test execution happening?
+  e2e_pool_definion: # where is e2e test execution happening?
+  stage_name: # needed to make job names unique if they are included multiple times
+  environment: # used to pass environment variables into the downstream scripts
+
+jobs:
+- job: compile_${{parameters.stage_name}}
+  condition: not(eq(variables['MODE'], 'e2e'))
```

Comment:
we may want to exclude all the stuff for custom triggered builds for now

Reply:
at the time of your review, I would have agreed. But I have spend the past few days fixing the end to end test execution, and I would suggest to keep the e2e mode in.
This would allow people to run the end to end tests at least in their private forks, and hopefully soon through CI bot as well.